modules/core/src/main/scala/org/tessellation/domain/cluster/programs/TrustPush.scala

```diff
@@ -1,30 +1,57 @@
 package org.tessellation.domain.cluster.programs
 
 import cats.effect.Async
+import cats.syntax.applicative._
 import cats.syntax.flatMap._
 import cats.syntax.functor._
 
-import org.tessellation.domain.trust.storage.TrustStorage
+import org.tessellation.domain.trust.storage.{SnapshotOrdinalTrustStorage, TrustStorage}
+import org.tessellation.schema.SnapshotOrdinal
 import org.tessellation.sdk.domain.gossip.Gossip
 
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.NonNegative
+import eu.timepit.refined.refineV
+import eu.timepit.refined.types.numeric.NonNegLong
+
 object TrustPush {
 
   def make[F[_]: Async](
     trustStorage: TrustStorage[F],
+    ordinalTrustStorage: SnapshotOrdinalTrustStorage[F],
     gossip: Gossip[F]
-  ): TrustPush[F] = new TrustPush[F](trustStorage, gossip) {}
+  ): TrustPush[F] = new TrustPush[F](trustStorage, ordinalTrustStorage, gossip) {}
 
 }
 
 sealed abstract class TrustPush[F[_]: Async] private (
   trustStorage: TrustStorage[F],
+  ordinalTrustStorage: SnapshotOrdinalTrustStorage[F],
   gossip: Gossip[F]
 ) {
+  val snapshotOrdinalTrustGossipInterval = NonNegLong(1000L)
+  var targetSnapshotOrdinal = SnapshotOrdinal.MinValue
```

Comment:
`var` should not be used for writing concurrent fashion code. please use `Ref` that gets concurrent access and referential transparency.

Reply:
Implemented this as a `Ref`.